### DISCUSSION TITLE
examples with allcfs command no response

### DISCUSSION THREAD

**Discussion by knmcguire** [USC-ACTLab/crazyswarm#555] [Q&A] [answered]
Hi!

I've noticed that the examples where you only use one line in the python script to execute an command does not seem to work.
Like this one in nicehover.py
`    allcfs.takeoff(targetHeight=Z, duration=1.0+Z)
`
This part of the code the crazyflie does react on:
    for cf in allcfs.crazyflies:
        pos = np.array(cf.initialPosition) + np.array([0, 0, Z])
        cf.goTo(pos, 0, 1.0)

But then it doesn't react on the land again with this:
`    allcfs.land(targetHeight=0.02, duration=1.0+Z)
`
This seems to happen even for just one crazyflie. 

So is this allcfs.command() api deprecated?

Update 13/12:

used the 2021.06 release of the crazyflie firmware

**Comment by whoenig** [comment]
This should work and I actually use the allcfs version almost exclusively. What firmware are you running? allcfs uses broadcast communication which (as far as I know) is currently not tested/supported within Bitcraze, so it is possible that the current nrf51/stm32 master branch is broken with respect to broadcasting.

**Reply by knmcguire** [reply]
Ah oke, thought so as it is excessively used within the examples!

I used the 2021.06 release of the stm/nrf firmware but saw the same thing I think on the latest of the STM as well (but will double check that). We have had some problems with the swarm library of the cflib of which we are not really sure of where the culprit is. But those are mostly instability issues.

Are you running la

**Reply by knmcguire** [reply]
Woops, accidently hit enter.

Are you using the latest firmware of the STM yourself?

**Reply by whoenig** [reply]
I mostly use 2021.06 with a few cherry-picked fixes (in particular the one for setting parameters by name; shouldn't be critical for your use-case, unless you try to set parameters using broadcasts). Did you upgrade your Crazyradio firmware? If you don't use it with motion capture it shouldn't matter, but I only test with the radio binary provided in the repo (which adds the option to "pack" two radio commands into a single USB request in some special cases). Unfortunately, there is currently no error/warning if you use a stock radio.

**Reply by knmcguire** [reply]
ah yeah that is not something I've tried myself no. I've been using the crazyradio as is so I'll give that a try.

I've tried it Lighthouse when I noticed this with the one crazyflie.

**Comment by uhvardhan** [comment]
I face the same issue as @knmcguire is facing. 

When I run niceHover.py - all CF's takeoff immediately and hover until i press a button after which it does land. 

I was also wondering if why only the last 2 CF's show up always in RVIZ? Is it a problem?

I have a video which I am unable to attach.

**Reply by uhvardhan** [reply]
Here is the link: 
https://drive.google.com/file/d/1FrrQtGKeR7y2RAtK02O2StD4_iOV3fvh/view?usp=sharing

**Reply by knmcguire** [reply]
Ah but your cfs do land based on the allcfs land command in the script right? This is the correct behavior since mine did not react and land after the script closed. So it's probably a different issue.

**Reply by uhvardhan** [reply]
Yes. It looks to be a slightly different issue.

**Comment by knmcguire** [comment, marked as answer]
Took a while for me to try it out but I found out what fixed it! For me the solution as to **update the crazyradio firmware.** 

When I did 
`rosrun crazyflie_tools scan -v`
 I got 
`Found Crazyradio with version 0.53`
Which is super old... And ofcourse I though when I read the i[nstructions to update all firmware](https://crazyswarm.readthedocs.io/en/latest/configuration.html#update-firmware), like: _nah how bad can it be?_ Lesson:... follow instructions ! :sweat_smile: 

So after following the instructions on the [crazyradio firmware readme](https://github.com/bitcraze/crazyradio-firmware) to update the crazyradio PA, it works. So your gut-feeling was correct @whoenig ! thanks!

**Reply by uhvardhan** [reply]
`Found Crazyradio with version 99.55`

This is the output when I do the `rosrun` as you did but I still do not have the expected behavior.

**Reply by knmcguire** [reply]
Are you sure you have the same problem as me? You mentioned in your last reaction that your crazyflie did react correctly on allcf.takeoff and land, while it did not with me before I updated the firmware.

The rviz not showing all cfs all the time is something I noticed as well, but that might be due to a slow downlink, which is a different issue. Perhaps its better to start a new discussion topic for that.

**Reply by uhvardhan** [reply]
Oh! Wait! I am sorry. 

I am having the opposite of your problem. 

`allcfs.takeoff` works for me but the loop doesn't.
